frontend/src/components/Connect/Details/Overview/Overview.tsx

```diff
@@ -20,35 +24,90 @@ const Overview: React.FC = () => {
 
   const { running, failed } = getTaskMetrics(tasks);
 
+  const hasTraceInfo = connector.status.trace;
+
+  const handleStateClick = () => {
+    if (connector.status.state === ConnectorState.FAILED && hasTraceInfo) {
```

Comment:
Looks like will be better to move this condition into separate helper. Something like   
  
```typescript
const canShowTrace = (connector: Connector) =>
  connector.status.state === ConnectorState.FAILED && !!connector.status.trace;
``` 

 And use it here firstly
```typescript
const handleStateClick = () => {
    if (canShowTrace(connector)) {
      setShowTraceModal(true);
    }
  };
```